### ISSUE TITLE
Crash To Desktop when click on calltip created with LUA or Python Script Plugin

### ISSUE THREAD

**Issue by Khundiann** [notepad-plus-plus/notepad-plus-plus#14664]
### Description of the Issue

When creating calltips with "Overload arrows" with either PythonScript or LuaScript plugins, clicking on the down arrow makes Notepad++ crash to desktop. Clicking on the up arrow or general area of the calltip works as it should.

### Steps to Reproduce the Issue

1. Install the PythonScript plugin
2. run the script below

```python
def showcalltip(args):
	calltip = "\001 Click \002"
	pos = editor.getCurrentPos()
	editor.callTipShow(pos, calltip)

def printclick(args):
    print(args)

console.show()
print("Test script running.")

editor.callback(showcalltip, [SCINTILLANOTIFICATION.CHARADDED])
editor.callback(printclick, [SCINTILLANOTIFICATION.CALLTIPCLICK])

```

3. Type any character in a document and the calltip will appear.
4. Click on the up arrow and the general area of the calltip and the arguments will be shown in console.
5. When clicking on the down arrow (or using the alt + down shortcut) makes Notepad++ crash to desktop with Exception Code: c0000094 which is a division by zero code. Seems the position key cannot return a value 2 as it fails to get a value. And Exception Code: c000041d a STATUS_FATAL_USER_CALLBACK_EXCEPTION code.

### Expected Behavior

When clicking the down arrow I expected to see the arguments in console.
### Actual Behavior

Notepad++ crashed to desktop
### Debug Information

```
Notepad++ v8.6   (64-bit)
Build time : Nov 23 2023 - 16:58:44
Path : P:\npp.8.6.portable.x64\notepad++.exe
Command Line : 
Admin mode : OFF
Local Conf mode : ON
Cloud Config : OFF
OS Name : Windows 10 Pro (64-bit)
OS Version : 22H2
OS Build : 19045.3930
Current ANSI codepage : 1252
Plugins : 
    ComparePlus (1.1)
    Explorer (1.9.9)
    GitSCM (1.4.8.2)
    HTMLTag (1.4.1)
    LuaScript (0.12)
    mimeTools (2.9)
    nppAutoDetectIndent (2.3)
    NppConverter (4.5)
    NppEventExec (0.9)
    NppExec (0.8.4)
    NppExport (0.4)
    NppSnippets (1.7.1)
    NppUISpy (1.2)
    NppXmlTreeviewPlugin (2)
    PreviewHTML (1.3.2)
    Python Indent (1.0.0.5)
    PythonScript (2)
    XMLTools (3.1.1.13)
    _CustomizeToolbar (5.3)
```

Exactly the same happens when using the LuaScript plugin, I'll include a test script for that too.

```lua
function showcalltip(ch)
	calltip = "\001 Click \002"
	pos = editor.CurrentPos
	editor:CallTipShow(pos, calltip)
    return false
end

function printclick(flag)
    print(flag)
    return false
end

print("Test script running.")
 
npp.AddEventHandler("OnChar", showcalltip)
npp.AddEventHandler("OnCallTipClick", printclick)

```

**Comment by alankilborn** [comment]
More discussion on the Community site: https://community.notepad-plus-plus.org/topic/25374

**Comment by Khundiann** [comment]
> I'm curious if you've tried this _minimally_, i.e., with only the plugins that ship with Notepad++, and a singled scripting plugin (i.e., if you're trying the Python, then only have the PythonScript plugin installed)

I just tried with a clean portable install of Notepad++ 8.6.2 and the result is the same.

Notepad++ v8.6.2   (64-bit)
Build time : Jan 14 2024 - 02:16:00
Path : P:\npp.8.6.2.portable.x64\notepad++.exe
Command Line : 
Admin mode : OFF
Local Conf mode : ON
Cloud Config : OFF
OS Name : Windows 10 Pro (64-bit)
OS Version : 22H2
OS Build : 19045.3930
Current ANSI codepage : 1252
Plugins : 
    mimeTools (3)
    NppConverter (4.5)
    NppExport (0.4)
    PythonScript (2)
    RandomValuesNppPlugin (0.3)

**Comment by rdipardo** [comment]
The crash does not occur when the buffer is normal text (i.e. has no active syntax highlighting).

@Khundiann, is that also the case for you?

**Comment by Khundiann** [comment]
> The crash does not occur when the buffer is normal text (i.e. has no active syntax highlighting).
> 
> @Khundiann, is that also the case for you?

@rdipardo I can confirm running the script and testing it with "normal text" selected as language works.

**Comment by xomx** [comment]
1st exception is the integer division by zero because of the `_currentNbOverloads == 0` there (then the 0xC000041D follows)

https://github.com/notepad-plus-plus/notepad-plus-plus/blob/f431a379671936ae611b8a231ee0a43c8a7fc0f9/PowerEditor/src/ScintillaComponent/FunctionCallTip.cpp#L108

**Comment by Khundiann** [comment]
@xomx @rdipardo That was solved swiftly and with style! Thanks guys!
Should I close this issue, or will someone else take care of that?

**Comment by alankilborn** [comment]
>Should I close this issue, or will someone else take care of that?

You shouldn't close anything.
Nothing is fixed yet.
If a fix gets made, part of the fix acceptance will include closing this issue.

**Comment by xomx** [comment]
@Khundiann 
Do not close.
I am not the author of this FunctionCallTip code, so at the moment I have no idea what is going on and why.
Maybe there is a problem or something unusual in your script, but of course the N++ should not crash even if that were true.

**Comment by donho** [comment]
@Khundiann 
Please check fix #14667 to see if your crash is fixed.

**Comment by Khundiann** [comment]
> @Khundiann Please check fix #14667 to see if your crash is fixed.

@donho 
I can confirm the fix works, Thanks!

![2024-02-03 20_47_38-Window](https://github.com/notepad-plus-plus/notepad-plus-plus/assets/151635111/e82466d6-2be6-4a15-b651-21b61984836d)

Notepad++ v8.6.2   (64-bit)
Build time : Feb  3 2024 - 20:32:33
Path : P:\PROJECTS VSCODE\notepad++\PowerEditor\bin64\notepad++.exe
Command Line : 
Admin mode : OFF
Local Conf mode : OFF
Cloud Config : OFF
OS Name : Windows 10 Pro (64-bit)
OS Version : 22H2
OS Build : 19045.3930
Current ANSI codepage : 1252
Plugins : 
    PythonScript (2)
